### PR TITLE
Optimize GitHub workflow

### PR DESCRIPTION
**Pull request by baschny**
* Add YAML Schema for PhpStorm
* Fix some headlines in the workflow file
* Make docker image artifacts and images names more understandable
* Output docker log before running the test that might fail (else the output is never shown anywhere)

### CODE REVIEW

**Comment by baschny** [comment]
@remuslazar I would merge this one first, because it will help me on the next PRs where the output is currently hidden and I understand the workflow better with the changes.

Both 7.2 and 7.3 still fail, but due to outside reasons:

- Neos 4.3 + 7.2 - requires composer v1 - will be fixed in a follow-up PR
- Neos 5.3 - fails due to the broken upgrade to "fluid" in the latest flow branch (not yet fixed upstream!)

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -1,48 +1,50 @@
...
-        run: docker save croneu/neos:${{ matrix.php-version }} | gzip > docker-neos-${{ matrix.target }}.tar.gz
+        name: "Save Docker Image"
+        run: docker save croneu/neos:${{ matrix.php-version }}-${{ matrix.target }} | gzip > docker-neos-${{ matrix.target }}.tar.gz
 
-      - name: Upload Docker Image as Artifact
+      - name: "Upload Docker Image as Artifact"
         uses: actions/upload-artifact@v2
         with:
-          name: docker-neos-${{ matrix.target }}-${{ matrix.php-version }}.tar.gz
+          name: docker-neos-${{ matrix.php-version }}-${{ matrix.target }}
```
Comment (remuslazar) [inline .github/workflows/main.yml:47]: The name is also being used for the artifacts, this was the reason I have used the right extension. Not using the gz extension will confuse `docker load`, e.g. when downloading the image from GitHub and loading it in docker for debugging.

File: .idea/jsonSchemas.xml [hunk 2]
```diff
@@ -0,0 +1,25 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
+  <component name="JsonSchemaMappingsProjectConfiguration">
+    <state>
+      <map>
+        <entry key="github-workflow">
+          <value>
+            <SchemaInfo>
+              <option name="name" value="github-workflow" />
+              <option name="relativePathToSchema" value="https://json.schemastore.org/github-workflow" />
```
Comment (remuslazar) [inline .idea/jsonSchemas.xml:10]: cool stuff! 👍

File: .github/workflows/main.yml (hunk 1 above)
Comment (baschny) [reply .github/workflows/main.yml:47]: Not really what I experienced: the "name" is used as the artifact name, when you download it, the download will be called "<name>.zip", so you get a file called "docker-neos-7.2.tar.gz.zip" which is simply a zip file containing the actual files (the ones specified in the "path", i.e. "docker-neos.tar.gz.").

Comment (remuslazar) [reply .github/workflows/main.yml:47]: ok, I just saw the name in the HTML link, never tried to download the actual file itself. I am cool with that when it works, maybe docker load will check the file header and not the file name.

**Pull request merged**